// storybook-addon-next/dist/images/webpack.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.configureImages = void 0;
const tslib_1 = require("tslib");
const semver_1 = (0, tslib_1.__importDefault)(require("semver"));
const utils_1 = require("../utils");
const configureImages = (baseConfig) => {
    configureStaticImageImport(baseConfig);
    (0, utils_1.addScopedAlias)(baseConfig, 'next/image');
};
exports.configureImages = configureImages;
const configureStaticImageImport = (baseConfig) => {
    var _a;
    const version = (0, utils_1.getNextjsVersion)();
    if (semver_1.default.lt(version, '11.0.0'))
        return;
    const rules = (_a = baseConfig.module) === null || _a === void 0 ? void 0 : _a.rules;
    rules === null || rules === void 0 ? void 0 : rules.forEach((rule, i) => {
        var _a;
        if (typeof rule !== 'string' &&
            rule.test instanceof RegExp &&
            rule.test.test('test.jpg')) {
            rules[i] = {
                test: rule.test,
                use: [
                    {
                        loader: require.resolve('./next-image-loader-stub'),
                        options: {
                            filename: (_a = rule.generator) === null || _a === void 0 ? void 0 : _a.filename
                        }
                    }
                ]
            };
        }
    });
};
